### DISCUSSION TITLE
Statistics by directory override language

### DISCUSSION THREAD

**Discussion by Soltus** [github-linguist/linguist#6299] [Classification] [answered]
At present, it seems to only support `*.in linguist-language=python`

I have a demand case，The .ipynb file in the /notebook/csharp directory should counted as C# ，then .ipynb file in the /notebook/fsharp directory should counted as F#

**Comment by lildude** [comment, marked as answer]
> I have a demand case，The .ipynb file in the /notebook/csharp directory should counted as C# ，then .ipynb file in the /notebook/fsharp directory should counted as F#

Linguist analyses files in isolation and doesn't consider the directory structor or other files within a directory at all. So if you want all the `.ipynb` files in `/notebook/csharp` to be considered as C# and those in ` /notebook/fsharp` to be F#, you will need to implement an override like this:

```gitattributes
notebook/csharp/**/*.ipynb linguist-language=csharp
notebook/fsharp/**/*.ipynb linguist-language=fsharp
```
